vector/src/main/java/im/vector/riotx/features/settings/VectorSettingsGeneralFragment.kt

```diff
@@ -370,26 +405,88 @@ class VectorSettingsGeneralFragment : VectorSettingsBaseFragment() {
      * Update the avatar.
      */
     private fun onUpdateAvatarClick() {
-        notImplemented()
+        if (!::avatarSelector.isInitialized) {
+            avatarSelector = AvatarSelectorView(activity!!, activity!!.layoutInflater, this)
+        }
+        mUserAvatarPreference.mAvatarView?.let {
+            avatarSelector.show(it, false)
+        }
+    }
 
-        /* TODO
-        if (checkPermissions(PERMISSIONS_FOR_TAKING_PHOTO, this, PERMISSION_REQUEST_CODE_LAUNCH_CAMERA)) {
-            changeAvatar()
+    override fun onTypeSelected(type: AvatarSelectorView.Type) {
+        when (type) {
+            AvatarSelectorView.Type.CAMERA  -> {
+                if (checkPermissions(PERMISSIONS_FOR_TAKING_PHOTO, this, PERMISSION_REQUEST_CODE_LAUNCH_CAMERA)) {
+                    avatarCameraUri = MultiPicker.get(MultiPicker.CAMERA).startWithExpectingFile(this)
+                }
+            }
+            AvatarSelectorView.Type.GALLERY -> {
+                MultiPicker.get(MultiPicker.IMAGE).single().startWith(this)
+            }
         }
-        */
     }
 
-    private fun changeAvatar() {
-        /* TODO
-        val intent = Intent(activity, VectorMediaPickerActivity::class.java)
-        intent.putExtra(VectorMediaPickerActivity.EXTRA_AVATAR_MODE, true)
-        startActivityForResult(intent, VectorUtils.TAKE_IMAGE)
-        */
+    private fun onAvatarSelected(image: MultiPickerImageType) {
+        val destinationFile = File(requireContext().cacheDir, "${image.displayName}_edited_image_${System.currentTimeMillis()}")
+        val uri = image.contentUri
+        UCrop.of(uri, destinationFile.toUri())
+                .withOptions(
+                        UCrop.Options()
```

Comment:
Didn't check, but if Ucrop options are the same than for attachment, we should extract this in a helper

Reply:
Maybe not the same that for attachment. We should limit to a square image, or if possible a round image